### PR TITLE
chore: CE change for moduleInstanceEditor route

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

- Add route changes for module instance editor
- Modify NameEditorComponent props to support module instance entity 
   - As the Module instance doesn't have pluginType, we make the NameEditorComponent independent of pluginType so the module instance editor could also use the same component.
 

Refer https://github.com/appsmithorg/appsmith-ee/pull for complete changes

#### PR fixes following issue(s)
Fixes #

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by ashit-rath** [comment]
@ankitakinger Adding you as reviewer since you recently made changes to the files in the PR and please advise with the split for the route as well

**Comment by ankitakinger** [comment]
@ashit-rath thanks for notifying me. 
@Rishabh-Rathod I think the other changes that you have done should be good in terms of the changes I have done till now. The only change I request you to make is about the split for routes. Please follow the `AppHeader` and `AppRouter` components for this split.

File: app/client/src/components/editorComponents/ActionNameEditor.tsx [hunk 1]
```diff
@@ -81,25 +80,32 @@ interface ActionNameEditorProps {
 function ActionNameEditor(props: ActionNameEditorProps) {
   const params = useParams<{ apiId?: string; queryId?: string }>();
 
-  const currentActionConfig: Action | undefined = useSelector(
```
Comment (ankitakinger) [inline app/client/src/components/editorComponents/ActionNameEditor.tsx:84]: why are we removing types here?

File: app/client/src/components/editorComponents/ActionNameEditor.tsx [hunk 2]
```diff
@@ -81,25 +80,32 @@ interface ActionNameEditorProps {
...
-      currentActionConfig={currentActionConfig}
       /**
        * This component is used by module editor in EE which uses a different
        * action to save the name of an action. The current callers of this component
        * pass the existing saveAction action but as fallback the saveActionName is used here
        * as a guard.
        */
       dispatchAction={props.saveActionName || saveActionName}
+      id={currentActionConfig?.id}
+      idUndefinedErrorMessage="Invalid Action id provided"
```
Comment (ankitakinger) [inline app/client/src/components/editorComponents/ActionNameEditor.tsx:106]: can we make this sense case and move it messages.ts?

File: app/client/src/pages/Editor/Explorer/Entity/Name.tsx [hunk 3]
```diff
@@ -149,8 +155,11 @@ export const EntityName = React.memo(
 
     return (
       <NameEditorComponent
-        currentActionConfig={{ id: props.entityId, name: updatedName }}
         dispatchAction={handleUpdateName}
+        id={props.entityId}
+        idUndefinedErrorMessage="No correct API id or Query id found in the url."
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/Explorer/Entity/Name.tsx:160]: can we make this sense case and move it messages.ts?

File: app/client/src/components/editorComponents/ActionNameEditor.tsx (hunk 2 above)
Comment (Rishabh-Rathod) [reply app/client/src/components/editorComponents/ActionNameEditor.tsx:106]: Do we also store the console error log messages in messages.ts? 
I believe that we only store the messages that we show in UI.

**Review by ankitakinger — DISMISSED** [review]

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**